src/hasher.rs

```diff
@@ -43,7 +50,7 @@ pub trait Hasher: Clone {
 
     /// This associated function takes a slice of bytes and returns a hash of it.
     /// Used by `concat_and_hash` function to build a tree from concatenated hashes
-    fn hash(data: &[u8]) -> Self::Hash;
+    fn hash(data: &[u8], hash_type: HashType) -> Self::Hash;
```

Comment:
I feel this definition is incorrect. Keccak IS a Hasher; so what would happen if we did:

```
keccak::hash(b"hello", HashType::SHA3)
```